pkg/experiment/metastore/index/query_test.go

```diff
@@ -265,3 +271,348 @@ func TestIndex_Query(t *testing.T) {
 		query(t, tx, idx)
 	})
 }
+
+func Test_QueryConcurrency(t *testing.T) {
+	for i := 0; i < 10 && !t.Failed(); i++ {
+		q := new(queryTestSuite)
+		q.run(t)
+	}
+}
```

Comment:
The test is quite intensive (tens of thousands of queries), but it should not take longer than a second on average. Running it locally for extended periods has not revealed any races or inconsistencies; if the test causes issues in CI, we should decrease the number of iterations.

Ideally, we should create a library of expensive tests to run in the background, with a shorter version – a smoke test – for the CI context.